[18.0][MIG] shopfloor: migrate `location_content_transfer` scenario

sebalix opened the pull request:
Depends on:
- ~https://github.com/OCA/stock-logistics-workflow/pull/2058~
- https://github.com/OCA/stock-logistics-workflow/pull/2076

Required a fix on `stock_storage_type` to make CI :green_circle: :
- https://github.com/OCA/stock-logistics-putaway/pull/12

Ref 5991

sebalix committed:
[MIG] shopfloor: migrate 'location_content_transfer' scenario

simahawk reviewed (commented):
LG, missing a pending test req apparently

simahawk commented:
These warnings should be fixed
```
 2025-08-06 15:54:55,545 404 WARNING odoo odoo.addons.base_rest.components.service: DEPRECATED: You must define an output schema for method cancel_work in service shopfloor.location.content.transfer 
2025-08-06 15:54:55,577 404 WARNING odoo odoo.addons.base_rest.components.service: DEPRECATED: You must define an output schema for method find_work in service shopfloor.location.content.transfer 
2025-08-06 15:54:55,703 404 WARNING odoo odoo.addons.base_rest.components.service: DEPRECATED: You must define an output schema for method find_work in service shopfloor.location.content.transfer 
2025-08-06 15:54:55,783 404 WARNING odoo odoo.addons.base_rest.components.service: DEPRECATED: You must define an output schema for method find_work in service shopfloor.location.content.transfer 
```
We have always a validator for the response but here.

grindtildeath committed:
shopfloor: Fix location_content_transfer tests

In case no backorder is created, there's no reason to update
the demand quantity, as it can still reflect what was the orginial
demand.

grindtildeath commented:
@simahawk I had a look at this and IMO the original test is wrong. I did a fix there: https://github.com/camptocamp/stock-logistics-shopfloor/pull/4

@jbaudoux do you agree?

simahawk commented:
I've updated the test requirements to drop 

https://github.com/OCA/stock-logistics-workflow/pull/2058

in favor of 

https://github.com/OCA/stock-logistics-workflow/pull/2076

simahawk commented:
> @simahawk I had a look at this and IMO the original test is wrong. I did a fix there: [camptocamp#4](https://github.com/camptocamp/stock-logistics-shopfloor/pull/4)
> 
> @jbaudoux do you agree?

Thanks, I included your fix now. Works locally!

simahawk commented:
Remaining errors
```
2025-08-21 11:12:18,095 337 INFO odoo odoo.sql_db: ConnectionPool(read/write;used=0/count=0/max=64): Closed 3 connections  

errors that caused failure (9):
2025-08-21 11:08:49,473 337 WARNING odoo odoo.addons.base_rest.components.service: DEPRECATED: You must define an output schema for method cancel_work in service shopfloor.location.content.transfer 
Error: Errors detected in log.
2025-08-21 11:08:49,506 337 WARNING odoo odoo.addons.base_rest.components.service: DEPRECATED: You must define an output schema for method find_work in service shopfloor.location.content.transfer 
2025-08-21 11:08:49,637 337 WARNING odoo odoo.addons.base_rest.components.service: DEPRECATED: You must define an output schema for method find_work in service shopfloor.location.content.transfer 
2025-08-21 11:08:49,722 337 WARNING odoo odoo.addons.base_rest.components.service: DEPRECATED: You must define an output schema for method find_work in service shopfloor.location.content.transfer 
2025-08-21 11:11:50,095 337 ERROR odoo odoo.addons.shopfloor_reception.tests.test_set_destination: FAIL: TestSetDestination.test_auto_posting
Traceback (most recent call last):
  File "/__w/stock-logistics-shopfloor/stock-logistics-shopfloor/shopfloor_reception/tests/test_set_destination.py", line 141, in test_auto_posting
    self.assertNotEqual(picking, selected_move_line.picking_id)
AssertionError: stock.picking(571,) == stock.picking(571,)
 
2025-08-21 11:11:57,919 337 ERROR odoo odoo.addons.shopfloor_reception.tests.test_set_quantity: FAIL: TestSetQuantity.test_concurrent_update_2
Traceback (most recent call last):
  File "/__w/stock-logistics-shopfloor/stock-logistics-shopfloor/shopfloor_reception/tests/test_set_quantity.py", line 724, in test_concurrent_update_2
    self.assertEqual(move_line_user_1.state, "done")
AssertionError: 'confirmed' != 'done'
- confirmed
+ done
```

simahawk commented:
/ocabot merge minor
